src/ADAL.PCL.Android/AuthenticationAgentContinuationHelper.cs

```diff
@@ -25,15 +25,31 @@ public static class AuthenticationAgentContinuationHelper
     {
         public static void SetAuthenticationAgentContinuationEventArgs(int requestCode, Result resultCode, Intent data)
         {
-            AuthorizationResult authorizationResult;
-            switch (resultCode)
+            AuthorizationResult authorizationResult = null;
+            switch ((int)resultCode)
             {
-                case Result.Ok: authorizationResult= new AuthorizationResult(AuthorizationStatus.Success, data.GetStringExtra("ReturnedUrl")); break;
-                case Result.Canceled: authorizationResult = new AuthorizationResult(AuthorizationStatus.UserCancel, null); break;
-                default: authorizationResult = new AuthorizationResult(AuthorizationStatus.UnknownError, null); break;
+                case (int)Result.Ok:
+                    authorizationResult = new AuthorizationResult(AuthorizationStatus.Success, data.GetStringExtra("ReturnedUrl"));
+                    break;
+
+                case (int)Result.Canceled:
+                    authorizationResult = new AuthorizationResult(AuthorizationStatus.UserCancel, null);
+                    break;
+
+                case 2001:
```

Comment:
Can you make all the result code constant value? it's hard to tell what does those mean.